com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/EvaluateRequestHandler.java

```diff
@@ -58,16 +56,16 @@
         String expression = evalArguments.expression;
 
         if (StringUtils.isBlank(expression)) {
-            throw AdapterUtils.createCompletionException(
+            throw new CompletionException(AdapterUtils.createUserErrorDebugException(
                 "Failed to evaluate. Reason: Empty expression cannot be evaluated.",
-                ErrorCode.EVALUATE_FAILURE);
+                ErrorCode.EVALUATION_COMPILE_ERROR));
         }
         StackFrameReference stackFrameReference = (StackFrameReference) context.getRecyclableIdPool().getObjectById(evalArguments.frameId);
         if (stackFrameReference == null) {
-            // stackFrameReference is null means the stackframe is continued by user manually,
-            throw AdapterUtils.createCompletionException(
-                "Failed to evaluate. Reason: Cannot evaluate because the thread is resumed.",
-                ErrorCode.EVALUATE_FAILURE);
+            // stackFrameReference is null means the given thread is running
+            throw new CompletionException(AdapterUtils.createUserErrorDebugException(
+                    "Failed to evaluate. Reason: Cannot evaluate because the thread is resumed.",
```

Comment:
=> `Failed to evaluate. Reason: Cannot evaluate because the thread is not suspended.`